packages/synapse-react-client/src/components/widgets/query-filter/EnumFacetFilter/EnumFacetFilterUI.tsx

```diff
@@ -0,0 +1,277 @@
+                    icon="close"
+                  />
+                </IconButton>
+              </Tooltip>
+            ),
+          }}
+          type="text"
+          placeholder="Find values"
+          value={searchTerm}
+          ref={textInput}
+          onChange={e => {
+            setSearchText(e.target.value)
+          }}
+        />
+        {!showSearch && (
+          <div className="EnumFacetFilter__checkAll">
+            <Checkbox
+              className="EnumFacetFilter__checkbox"
+              onChange={() => {
+                if (!allIsSelected) {
+                  onRemoveAllFacetSelections()
+                }
+              }}
+              key="select_all"
+              checked={allIsSelected}
+              label="All"
+              isSelectAll={true}
+            />
+            <Tooltip title={'Search'}>
+              <IconButton
+                size={'small'}
+                className="EnumFacetFilter__searchbtn"
+                onClick={() => {
+                  setSearchText('')
+                  setShowSearch(true)
+                  textInput.current?.focus()
+                }}
+              >
+                <IconSvg
+                  sx={{ fontSize: 'inherit' }}
+                  wrap={false}
+                  icon="search"
+                />
+              </IconButton>
+            </Tooltip>
+          </div>
+        )}
+      </div>
+      <div>
+        {facetValuesToShow.map((facetValueAndCount, index: number) => {
+          const { isSelected, displayText, value, count } = facetValueAndCount
+
+          return (
+            <EnumFacetFilterOption
+              key={value}
+              id={[
+                removeWhitespace(facetTitle),
+                removeWhitespace(value),
+                index,
+              ].join('-')}
+              label={displayText}
+              count={count}
+              isDropdown={isDropdown}
+              checked={isSelected}
+              onChange={(newIsSelected: boolean) => {
+                if (newIsSelected) {
+                  onAddValueToSelection(value)
+                } else {
+                  onRemoveValueFromSelection(value)
+                }
+              }}
+            />
+          )
+        })}
+        {!isDropdown && (
+          <>
+            {!showAll &&
+              facetValues.length > MAX_ENUMERATION_VALUES_TO_SHOW && (
+                <button
+                  className="EnumFacetFilter__showMoreFacetsBtn"
+                  onClick={() => setToggleShowAll(true)}
+                >
+                  <div className="EnumFacetFilter__checkboxContainer">
+                    <div className="EnumFacetFilter__showMoreFacetsLabel">
+                      Show all ({facetValues.length})
+                    </div>
+                  </div>
+                </button>
+              )}
+            {showAll &&
+              !searchTerm &&
+              !valueIndexedGreaterThanMaxToShowIsSelected &&
```

Comment:
Fixed an issue where
1. `showAll` is true because the e.g. 6th item is selected
2. The "Show less" button was shown
3. When clicked, "Show less" would do nothing

Now, hide the button if `valueIndexedGreaterThanMaxToShowIsSelected` is true.